prov/shm/src/smr_rma.c

```diff
@@ -159,9 +163,30 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 		}
 		resp = ofi_cirque_tail(smr_resp_queue(ep->region));
 		pend = freestack_pop(ep->pend_fs);
-		smr_format_iov(cmd, smr_peer_addr(ep->region)[peer_id].addr,
-			       iov, iov_count, total_len, op, 0, data,
-			       op_flags, context, ep->region, resp, pend);
+		if (ep->region->cma_cap) {
+			smr_format_iov(cmd, smr_peer_addr(ep->region)[peer_id].addr,
+				       iov, iov_count, total_len, op, 0, data,
+				       op_flags, context, ep->region, resp, pend);
+		} else {
+			/*
+			 * mmap protocol
+			 * Once SAR protocol gets merged, we need set a threshold
+			 * for switching from SAR to mmap
+			 */
+			msg_id = (ep->msg_id)++;
```

Comment:
We declare and pass this in? Can we move this increment into the format call?

Reply:
Yeah. I moved this into the combined format_mmap call.